system/mount.py

```diff
@@ -306,16 +321,59 @@ def umount(module, **kwargs):
     else:
         return rc, out+err
 
+
+# Note if we wanted to put this into module_utils we'd have to get permission
+# from @jupeter -- https://github.com/ansible/ansible-modules-core/pull/2923
+# @jtyr -- https://github.com/ansible/ansible-modules-core/issues/4439
+# and @abadger to relicense from GPLv3+
+def is_bind_mounted(module, dest, src=None, fstype=None):
+    """Return whether the dest is bind mounted
+    :arg module: The AnsibleModule (used for helper functions)
+    :arg dest: The directory to be mounted under.  This is the primary means
+        of identifying whether the destination is mounted.
+    :kwarg src: The source directory.  If specified, this is used to help
+        ensure that we are detecting that the correct source is mounted there.
+    :kwarg fstype: The filesystem type.  If specified this is also used to
+        help ensure that we are detecting the right mount.
+    :returns: True if the dest is mounted with src otherwise False.
+    """
+    bin_path = module.get_bin_path('mount', required=True)
+    cmd = '%s -l' % bin_path
+
+    if platform.system() == 'Linux':
+        # If Linux, findmnt works more reliably.  Some distros won't have
+        # findmnt and mount -l will still work, however.  It depends on
+        # whether the distro saves the source directory in the mtab
+        bin_path = module.get_bin_path('findmnt', required=True)
+        cmd = '%s -nr %s' % (bin_path, dest)
+
+    rc, out, err = module.run_command(cmd)
+    allmounts = out.split('\n')
+
+    for mounts in allmounts[:-1]:
+        arguments = mounts.split()
+
+        if platform.system() == 'Linux':
+            source = LINUX_BIND_MOUNT_RE.search(arguments[1]).group(1)
+            if arguments[0] == dest and (source == src or src is None):
+                return True
```

Comment:
That can be a thing in fstab but I don't think it's possible in mount and findmnt.  fstab is just a text file whereas mount and findmnt are reflecting state given by the kernel.  (Unless there's some implementations that use a little indirection with mtab and mtab is itself a real file rather than a pointer into kernel space on that implementation).  I know the latter is individually true but I don't know about the combination of the two.